common/modules/nft_deposit/src/nft_deposit.rs

```diff
@@ -0,0 +1,127 @@
+#![no_std]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use common_structs::{GenericEsdtAmountPair, Nonce};
+
+#[elrond_wasm_derive::module]
+pub trait NftDepositModule: token_send::TokenSendModule {
+    #[payable("*")]
+    #[endpoint(depositToken)]
+    fn deposit_token(
+        &self,
+        #[payment_token] payment_token_id: TokenIdentifier,
+        #[payment_nonce] payment_token_nonce: Nonce,
+        #[payment_amount] payment_amount: Self::BigUint,
+    ) -> SCResult<()> {
+        require!(payment_amount != 0, "Cannot deposit 0 tokens");
```

Comment:
required token id of what? this is a generic module. each module can have its own logic of require! outside this code

Reply:
this function has an endpoint - the function is public. it can be called from the outside, as I see right now. And if it can be called from the outside, it needs protection.

Reply:
done